tokio/src/io/read_buf.rs

```diff
@@ -283,3 +280,15 @@ impl fmt::Debug for ReadBuf<'_> {
             .finish()
     }
 }
+
+fn slice_to_uninit_mut(slice: &mut [u8]) -> &mut [MaybeUninit<u8>] {
+    unsafe {slice::from_raw_parts_mut::<MaybeUninit<u8>>(slice.as_mut_ptr().cast(), slice.len())}
+}
```

Comment:
This is unsafe too. The `&mut [MaybeUninit<u8>]` type can be used to write uninitialized bytes to the slice.

Reply:
@Darksonn can you please elaborate? How can one write uninit bytes into a `&mut [MaybeUninit<u8>]`?

I do think the code should add a comment (out of diligence) saying this is safe because `MaybeUninit` is spec'd to have same layout, alignment and ABI as T (`u8` in this case).

Reply:
A miri example is available [here][1].

[1]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1b281796db9c4ad5fddee1047fb4fd1b

Reply:
Ah I see what you mean.  Looks like miri is more "clever" than I anticipated :).  While I get the formalism of "this value is undefined", it's a bit strange to have such a notion for a `u8` given every bit representation is a valid value.

Reply:
Oh, right, why didn't I realize that.